narwhals/_arrow/dataframe.py

```diff
@@ -285,8 +286,18 @@ def join(
             ),
         )
 
-    def drop(self, *columns: str) -> Self:
-        return self._from_native_dataframe(self._native_dataframe.drop(list(columns)))
+    def drop(self: Self, *columns: str, strict: bool = True) -> Self:
+        cols = set(self.columns)
+        to_drop = list(columns)
+        if strict:
+            for d in to_drop:
+                if d not in cols:
+                    msg = f'"{d}" not found'
+                    raise ColumnNotFoundError(msg)
```

Comment:
The logic is same everywhere. I wonder if we should move it directly to `BaseFrame`.

Also, I noticed that polars is "greedy" in the raise, meaning that if 2 columns are in the drop list but missing, the error will only mention the first one

Reply:
I think Polars changed behaviour here in 1.0 too (before it would just ignore missing columns)

so, fine by me to just implement it in BaseFrame and let everyone benefit

Reply:
Cool, I just noticed CI failing for polars pre 1. Refactoring into base now :)

Reply:
Somehow I keep f*cking up parsing `*columns` - it's probably just late 🙈

Reply:
Ok it is somehow fixed but worth taking a double look at it as the expected types along the chain of calls are a bit different